File: 프로그래머스/lv2/87946. 피로도/피로도.py
from itertools import permutations as p
def solution(k, dungeons):
    m = 0
    l = len(dungeons)
    for o in p(range(l)):
        tk = k
        for n, i in enumerate(o):
            if tk < dungeons[i][0]:
                m = max(m, n)
                break
            tk -= dungeons[i][1]
        else:
            return l
    return m

File: 프로그래머스/lv2/87946. 피로도/피로도.js
function solution(k, dungeons) {
  const visited = new Array(dungeons.length).fill(false);
  let maxCnt = 0;

  function dfs(curK, curCnt) {
    if (curK < 0) return;
    maxCnt = Math.max(maxCnt, curCnt);
    for (let i = 0; i < dungeons.length; i++) {
      if (!visited[i] && curK >= dungeons[i][0]) {
        visited[i] = true;
        dfs(curK - dungeons[i][1], curCnt + 1);
        visited[i] = false;
      }
    }
  }

  dfs(k, 0);
  return maxCnt;
}